src/runit-pause.c: Add

From: Rajiv M Ranganath <rajiv.ranganath@atihita.com>

Imported from Void Linux's - `void-runit`
---
 src/runit-pause.c |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)
 create mode 100644 src/runit-pause.c

diff --git a/src/runit-pause.c b/src/runit-pause.c
new file mode 100644
index 0000000..4825df6
--- /dev/null
+++ b/src/runit-pause.c
@@ -0,0 +1,20 @@
+#include <unistd.h>
+#include <signal.h>
+
+static void
+nop(int sig)
+{
+}
+
+int
+main()
+{
+  signal(SIGTERM, nop);
+  signal(SIGINT, nop);
+  signal(SIGHUP, SIG_IGN);
+
+  pause();
+
+  return 0;
+}
+
